includes/Services/HashService.php

```diff
@@ -3,55 +3,141 @@
 namespace PowerBoard\Services;
 
 use PowerBoard\PowerBoardPlugin;
+use phpseclib\Crypt\AES;
+use phpseclib\Crypt\Base;
 
 class HashService {
-	private const CIPHER = 'AES-128-CBC';
+
+	private const CIPHER = 'aes-128-cbc';
 	private const OPTION = OPENSSL_RAW_DATA;
+	private const NONCE_LENGTH = 24;
+	private const KEY_LENGTH = 32;
 
 	public static function encrypt( string $string ): string {
 
-		if ( ! function_exists( 'openssl_encrypt' ) ) {
-			return base64_encode( $string );
-		}
+		if ( function_exists( 'sodium_crypto_secretbox' ) ) {
```

Comment:
@andryes How did you decide the ordering of these? Wouldn't openssl be the most reliable?

Reply:
@andryes Also what are the chances of "sodium_crypto", "phpseclib\Crypt\AES", "openssl_encrypt" - if we are adding a library it will always be there?